### DISCUSSION TITLE
Need to control the next request on collection

### DISCUSSION THREAD

**Discussion by manohjRM** [thunderclient/thunder-client-support#1137] [Q&A]
postman.setNextRequest() - How can I implement the same in TC custom filters? Please guide me.

Consider I have 3 request A,B and C.
I need to skip the request B based on the test result of A and need to run request C

Thanks!

**Comment by rangav** [comment]
Hi @manohjRM 

you can try add Request B as a pre request for Request B  in the `Pre Run` tab and see if the conditions will be useful to you.

**Reply by manohjRM** [reply]
Hi @rangav , I tried the same before, but I can't see the Request B under Pre Requests of Request B. Is there any way other that?

**Reply by manohjRM** [reply]
![image](https://user-images.githubusercontent.com/55882438/235336841-f2be5500-cf06-4fcb-bce9-6bcd187706bf.png)

**Reply by rangav** [reply]
You need to open `request A` to view request B
In pre requests

**Reply by manohjRM** [reply]
Actually, I need to run request A first and then I need to run request B or C. If I mention request B in request A, then req B will run before req A.

I need to mention the req B in the Test of req A.
